engine/access/rest/converter.go

```diff
@@ -3,21 +3,201 @@ package rest
 import (
 	"encoding/base64"
 	"fmt"
+	"regexp"
 
 	"github.com/onflow/flow-go/engine/access/rest/generated"
 	"github.com/onflow/flow-go/model/flow"
 )
 
-// Converter provides the conversion function to/from the Swagger object to Flow objects
+// Converter provides functionality to convert from request models generated using
+// open api spec and flow models.
 
-func toBlock(flowBlock *flow.Block) *generated.Block {
+// Flow section - converting request data to flow models with validation.
+
+func toID(id string) (flow.Identifier, error) {
+	valid, _ := regexp.MatchString(`^[0-9a-fA-F]{64}$`, id)
+	if !valid {
+		return flow.Identifier{}, fmt.Errorf("invalid ID")
+	}
+
+	return flow.HexStringToIdentifier(id)
+}
+
+func toAddress(address string) (flow.Address, error) {
+	valid, _ := regexp.MatchString(`^[0-9a-fA-F]{16}$`, address)
+	if !valid {
+		return flow.Address{}, fmt.Errorf("invalid address")
+	}
+
+	return flow.HexToAddress(address), nil
+}
+
+func toProposalKey(key *generated.ProposalKey) (flow.ProposalKey, error) {
+	address, err := toAddress(key.Address)
+	if err != nil {
+		return flow.ProposalKey{}, err
+	}
+
+	return flow.ProposalKey{
+		Address:        address,
+		KeyIndex:       uint64(key.KeyIndex),
+		SequenceNumber: uint64(key.SequenceNumber),
+	}, nil
+}
+
+func toSignature(signature string) ([]byte, error) {
+	signatureBytes := []byte(signature)
+	// the maximum signature length is arbitrarily chosen to be 2048. This check is to ensure that the signature is not
+	// absurdly long
+	if len(signatureBytes) > 2048 {
+		return nil, fmt.Errorf("invalid signature")
```

Comment:
Again, maybe adding detail would be nice.

Also, define a constant for 2048?

Reply:
added const for 2048

Reply:
I believe binary signatures are supposed to be 64bytes, so len could be the base 64 equivalent of that

Reply:
yup ur right ..i just checked [here](https://github.com/onflow/flow-go/blob/7054d07329903c999e6cdceb226292deebfa705d/crypto/types.go)